### DISCUSSION TITLE
Simple MQTT client with PyQt5

### DISCUSSION THREAD

**Discussion by azat385** [empicano/aiomqtt#319] [Q&A]
How to use aiomqtt for a simple mqtt client with PyQt5?

<img src="https://camo.githubusercontent.com/63113ce3fbfe119d701ae665aee16f5e9df3fc69a908b13bbc30b5bd654869b6/68747470733a2f2f692e696d6775722e636f6d2f7a7079316c78682e706e67" width="400">

The [sync code](https://github.com/iqfareez/py-mqttclient-app/blob/master/main.py) using paho is straightforward and easy to modify.
I have read [Async IO in Python](https://realpython.com/async-io-python/) several times, but couldnt get it running.

Please help to implement aiomqtt.

**Comment by JonathanPlasse** [comment]
What do you want to achieve with aiomqtt that paho-mqtt cannot?
Correct me if I am wrong but PyQt5 does not seem to be very async friendly.

**Reply by azat385** [reply]
PyQt5 is not async friedly at all )
I need to rewrite my cli scripts which were mainly based on callbacks with GUI and make the logic more complex. 
Previous callbacks become callback hell, so i decided that it will be easier to write complex logic with async code. am I right?

**Reply by azat385** [reply]
maybe i will shift to PySide6 with trio or async according to [Async “Minimal” Example](https://doc.qt.io/qtforpython-6/examples/example_async_minimal.html)

**Reply by JonathanPlasse** [reply]
I would recommend keeping using paho-mqtt.
Introducing async code, which does not mesh well with the rest of the code (PyQt5), will surely be more complex.
Do you have even non-working code where the callbacks become too complex?
It may be possible to rewrite it another way with a manageable complexity using paho-mqtt.
